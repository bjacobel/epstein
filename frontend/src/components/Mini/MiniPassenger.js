import React from 'react';
import { Link } from 'react-router-dom';
import { gql, useQuery } from '@apollo/client';
import classNames from 'classnames';

import ErrorBoundary from '../ErrorBoundary';
import Loading from '../Loading';
import { box, link, row, name, passengerBox } from './mini.css';
import { historyText } from '../../containers/Passenger';

export const PASSENGER = gql`
  query Passenger($slug: String!) {
    passenger(slug: $slug) {
      id
      name
      slug
      biography
      flightCount
      histogram {
        count
        month
      }
    }
  }
`;

export default ({ slug }) => {
  const { loading, error, data } = useQuery(PASSENGER, {
    variables: { slug },
  });

  if (loading) return <Loading />;
  if (error) throw error;

  return (
    <ErrorBoundary>
      <div className={classNames(box, passengerBox)}>
        <Link to={`/passenger/${data.passenger.slug}`} className={link}>
          <div className={classNames(row, name)}>{data.passenger.name}</div>
          <div className={row}>{data.passenger.biography}</div>
          <div className={row}>{historyText(data)}</div>
        </Link>
      </div>
    </ErrorBoundary>
  );
};
