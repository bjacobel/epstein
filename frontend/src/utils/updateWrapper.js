import { act } from 'react-dom/test-utils';

export function wait(amount = 0) {
  return new Promise(resolve => setTimeout(resolve, amount));
}

export default async (wrapper, amount = 0) => {
  await act(async () => {
    jest.useRealTimers();
    await wait(amount);
    jest.useFakeTimers();
    wrapper.update();
  });
};
