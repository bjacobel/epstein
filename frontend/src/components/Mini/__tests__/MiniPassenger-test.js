import React from 'react';
import { ApolloProvider, ApolloClient, InMemoryCache } from '@apollo/client';
import { mount } from 'enzyme';

import MockLink from '../../../utils/testing/MockLink';
import updateWrapper from '../../../utils/testing/updateWrapper';
import MiniPassenger, { PASSENGER } from '../MiniPassenger';

const data = {
  passenger: {
    id: 1,
    slug: 'bill-clinton',
    name: 'Bill Clinton',
    biography: 'prez',
    flightCount: 26,
    histogram: [
      { month: '2002-02-01', count: 2 },
      { month: '2002-03-01', count: 4 },
    ],
  },
};

describe('MiniPassenger', () => {
  let client, link;

  beforeEach(() => {
    link = new MockLink();
    client = new ApolloClient({
      cache: new InMemoryCache({
        addTypename: false,
      }),
      link,
    });
  });

  it('matches snapshot', async () => {
    const queryHandler = jest.fn().mockResolvedValue({ data });
    link.setRequestHandler(PASSENGER, queryHandler);

    const wrapper = mount(
      <ApolloProvider client={client}>
        <MiniPassenger slug="bill-clinton" />
      </ApolloProvider>,
    );
    await updateWrapper(wrapper);

    expect(wrapper.find(MiniPassenger)).toMatchSnapshot();
  });
});
