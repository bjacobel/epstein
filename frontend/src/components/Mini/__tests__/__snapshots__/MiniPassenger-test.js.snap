// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiniPassenger matches snapshot 1`] = `
<_default
  slug="bill-clinton"
>
  <ErrorBoundary>
    <div
      className="box passengerBox"
    >
      <Link
        className="link"
        to="/passenger/bill-clinton"
      >
        <a
          href="/passenger/bill-clinton"
        >
          <div
            className="row name"
          >
            Bill Clinton
          </div>
          <div
            className="row"
          >
            prez
          </div>
          <div
            className="row"
          >
            Appears in the Epstein flight logs 26 times between February and March 2002
          </div>
        </a>
      </Link>
    </div>
  </ErrorBoundary>
</_default>
`;
